src/content/en/updates/posts/2016/06/webrtc-ecdsa.markdown

```diff
@@ -0,0 +1,69 @@
+---
+layout: updates/post
+title: "ECDSA for WebRTC: better security, better privacy and better performance"
+description: "From version 52, Chrome uses ECDSA — a much more efficient and secure algorithm for WebRTC certificate key generation. In addition, RTCCertificates can now be stored with IndexedDB."
+published_on: 2016-06-17
+updated_on: 2016-06-17
+authors:
+  - samdutton
+tags:
+  - chrome52
+  - media
+  - security
+  - webrtc
+featured_image: /web/updates/images/2016/06/webrtc-ecdsa/featured.png
+---
+
+<style>
+.screenshot-landscape {
+ max-width: 60%;
+}
+.screenshot-portrait {
+ max-width: 35%;
+}
+@media screen and (max-width: 500px) {
+  img.screenshot {
+    max-width: 100%;
+  }
+}
+</style>
+
+<p class="intro">You can now choose which type of cryptographic certificate to use for setting up a WebRTC peer connection.</p>
+
+Until recently, WebRTC used RSA-1024 keys for certificates. There are several disadvantages with these keys:
+
+* Generating RSA-1024 keys adds around 1000ms to call setup time.
+* 1024-bit RSA keys do not provide adequate cryptographic strength.
+
+Because certificate generation with RSA-1024 is slow, some sites have resorted to preparing certificates in advance or reusing them.
+
+The key strength issue could be resolved by going to 2048-bit RSA keys or more, but that would delay call setup by several additional seconds. Instead of changing the RSA key size, Chrome 52 implements ECDSA keys (Elliptic Curve Digital Signature Algorithm) for use in certificates. These are as strong as 3072-bit RSA keys‚ but several thousand times faster: call setup overhead with ECDSA is just a few milliseconds.
+
+> Breaking an RSA key requires you to factor a large number. We are pretty good at factoring large numbers and [getting better all the time](http://bristolcrypto.blogspot.co.uk/2013/02/discrete-logarithms.html). Breaking an ECDSA key requires you to solve the Elliptic Curve Discrete Logarithm Problem (ECDLP). The mathematical community has not made any major progress in improving algorithms to solve this problem since is was independently introduced by Koblitz and Miller in 1985.
+
+— [Nick Sullivan](https://blog.cloudflare.com/ecdsa-the-digital-signature-algorithm-of-a-better-internet/), CloudFlare
+
+All in all, ECDSA keys mean better security, better privacy and better performance — especially on mobile. For these reasons, ECDSA has been mandated in the [WebRTC Security Architecture](https://www.ietf.org/mail-archive/web/rtcweb/current/msg14754.html) draft.
+
+From Chrome 47 you can opt in to ECDSA:
+
+{% highlight javascript %}
+pc.generateCertificate({name: "ECDSA", namedCurve: "P-256"})
+{% endhighlight %}
+
+From Chrome 52, ECDSA is enabled by default.
```

Comment:
Thanks — added.

On 21 June 2016 at 13:48, Harald Alvestrand <notifications@github.com>
wrote:

> In src/content/en/updates/posts/2016/06/webrtc-ecdsa.markdown
> <https://github.com/google/WebFundamentals/pull/3005#discussion_r67853375>
> :
>
> > +
> > +The key strength issue could be resolved by going to 2048-bit RSA keys or more, but that would delay call setup by several additional seconds. Instead of changing the RSA key size, Chrome 52 implements ECDSA keys (Elliptic Curve Digital Signature Algorithm) for use in certificates. These are as strong as 3072-bit RSA keys‚ but several thousand times faster: call setup overhead with ECDSA is just a few milliseconds.
> > +
> > +> Breaking an RSA key requires you to factor a large number. We are pretty good at factoring large numbers and [getting better all the time](http://bristolcrypto.blogspot.co.uk/2013/02/discrete-logarithms.html). Breaking an ECDSA key requires you to solve the Elliptic Curve Discrete Logarithm Problem (ECDLP). The mathematical community has not made any major progress in improving algorithms to solve this problem since is was independently introduced by Koblitz and Miller in 1985.
> > +
> > +— [Nick Sullivan](https://blog.cloudflare.com/ecdsa-the-digital-signature-algorithm-of-a-better-internet/), CloudFlare
> > +
> > +All in all, ECDSA keys mean better security, better privacy and better performance — especially on mobile. For these reasons, ECDSA has been mandated in the [WebRTC Security Architecture](https://www.ietf.org/mail-archive/web/rtcweb/current/msg14754.html) draft.
> > +
> > +From Chrome 47 you can opt in to ECDSA:
> > +
> > +{% highlight javascript %}
> > +pc.generateCertificate({name: "ECDSA", namedCurve: "P-256"})
> > +{% endhighlight %}
> > +
> > +From Chrome 52, ECDSA is enabled by default.
>
> http://w3c.github.io/webrtc-pc/#dom-rtcpeerconnection-generatecertificate
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/WebFundamentals/pull/3005/files/cc723712f20cb7a7045f8a7d2a8cbc4e1633178b#r67853375>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMhqgXxcuczJDUI1Jq_6uMla7iqKAhKks5qN890gaJpZM4I3jfz>
> .
>